test/unit/utils/api-request.spec.ts

```diff
@@ -450,6 +446,71 @@ describe('AuthorizedHttpClient', () => {
     });
   });
 
+  describe('HTTP Agent', () => {
+    let transportSpy: sinon.SinonSpy = null;
+    let mockAppWithAgent: FirebaseApp;
+    let agentForApp: Agent;
+
+    beforeEach(() => {
+      const options = mockApp.options;
+      options.httpAgent = new Agent();
+      const https = require('https');
+      transportSpy = sinon.spy(https, 'request');
+      mockAppWithAgent = mocks.appWithOptions(options);
+      agentForApp = options.httpAgent;
+    });
+
+    afterEach(() => {
+      transportSpy.restore();
+      transportSpy = null;
+      return mockAppWithAgent.delete();
+    });
+
+    it('should use the HTTP agent set in request', () => {
```

Comment:
Is it any different than 'should use the HTTP agent set in request' above. They look pretty similar.

Reply:
The first one was for the `HttpClient` class. This one's for `AuthorizedHttpClient`, which is a subclass of the former. The child class has the ability to use the agent configured at app-level, but also supports overriding it at request-level. These tests verify this.